tests/integration/targets/s3_bucket/roles/s3_bucket/tasks/ownership_controls.yml

```diff
@@ -0,0 +1,98 @@
+---
+- module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+
+    # ============================================================
+
+    - name: 'Create a simple bucket bad value for ownership controls'
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: present
+        object_ownership: default
+      ignore_errors: true
+      register: output
+
+    - assert:
+        that:
+          - output.failed
+
+    - name: 'Create bucket with object_ownership set to object_writer'
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: present
+      ignore_errors: true
+      register: output
+
+    - assert:
+        that:
+          - output.changed
+          - not output.object_ownership|bool
+
+    - name: delete s3 bucket
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: absent
+
+    - name: 'create s3 bucket with object ownership controls'
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: present
+        object_ownership: ObjectWriter
+      register: output
+
+    - assert:
+        that:
+          - output.changed
+          - output.object_ownership
+          - output.object_ownership == 'ObjectWriter'
+
+    - name: 'update s3 bucket ownership controls'
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: present
+        object_ownership: BucketOwnerPreferred
+      register: output
+
+    - assert:
+        that:
+          - output.changed
+          - output.object_ownership
+          - output.object_ownership == 'BucketOwnerPreferred'
+
+    - name: 'test idempotency update s3 bucket ownership controls'
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: present
+        object_ownership: BucketOwnerPreferred
+      register: output
+
+    - assert:
+        that:
+          - output.changed is false
+          - output.object_ownership
+          - output.object_ownership == 'BucketOwnerPreferred'
+
+    - name: 'delete s3 bucket ownership controls'
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: present
+        delete_object_ownership: true
+      register: output
+
+    - assert:
+        that:
+          - output.changed
+          - not output.object_ownership|bool
+
+  # ============================================================
+  always:
```

Comment:
I think we'll want a task in the `always` block to `delete_object_ownership` before deleting the bucket, in case one of the previous tasks fails with the ownership in an unexpected state.

Reply:
@jillr thanks for the feedback.
updated